backend/database.js

```diff
@@ -186,18 +178,22 @@ export default async () => {
     })
 
     sbp('sbp/selectors/overwrite', {
-      'chelonia/db/get': async function (key: string): Promise<Buffer | string | void> {
-        const lookupValue = cache.get(key)
+      'chelonia/db/get': async function (prefixableKey: string): Promise<Buffer | string | void> {
+        const lookupValue = cache.get(prefixableKey)
         if (lookupValue !== undefined) {
           return lookupValue
         }
-        const value = await readData(key)
-        if (value !== undefined) {
-          cache.set(key, value)
+        const [prefix, key] = parsePrefixableKey(prefixableKey)
+        let value = await readData(key)
+        if (value === undefined) {
+          return
         }
+        value = prefixHandlers[prefix](value)
+        cache.set(prefixableKey, value)
         return value
       },
       'chelonia/db/set': async function (key: string, value: Buffer | string): Promise<void> {
+        checkKey(key)
```

Comment:
So, we need to decide whether to have this here or have this somewhere else. At the moment, `readData` in `fs` the `checkKey` is still in `readData`, so that's an inconsistency. Should we have this here, or "closer to the action"?

Whatever is decided, it needs to be done consistently everywhere. For example, right now in the in-memory `'chelonia/db/get'` and `set`, it's not called. But it should be if we decided that it's better to have it here than in `readData` / `writeData`.

Reply:
Yea I was trying to improve consistency and performance, and thought maybe it'd be better to have one key check in `'chelonia/db/set'` to be sure no bad key can enter the DB, no matter what every individual backend's `writeData()` would do. Then the key check was no longer needed when getting or deleting entries, since the DB could now be assumed free from any invalid key... Except that in `fs` we also have to prevent path attacks, so I added a key check in `fs`'s `readData()` (actually I could have used a simpler check than checkKey here) . If there was a `deleteData()` function it would need the same key check.

Reply:
@snowteamer let me think on this. It might be fine.